File: C#/AddTwoNumbers.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    
    public ListNode AddTwoNumbers(ListNode l1, ListNode l2) {
        
        int sum = l1.val + l2.val;
            
        if (sum >= 10)
        {
            sum -= 10;
            if (l1.next != null)
            {
                l1.next.val += 1;   
            }
            else
            {
                l1.next = new ListNode(1, null);
            }
        }
        
        ListNode output = new ListNode(sum);
        
        if ((l1.next != null) || (l2.next != null))
        {
            if (l1.next == null)
            {
                l1.next = new ListNode(0, null);
            }
            if (l2.next == null)
            {
                l2.next = new ListNode(0, null);
            }
            
            output.next = AddTwoNumbers(l1.next, l2.next);   
        }
        
        return output;    
    }
}

File: Python/addTwoNumbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:

        valL1 = []
        valL2 = []

        while l1:
            valL1.append(l1.val)
            l1 = l1.next

        while l2:
            valL2.append(l2.val)
            l2 = l2.next

        while len(valL1) < len(valL2):
            valL1.append(0)

        while len(valL2) < len(valL1):
            valL2.append(0)

        index = 0
        remainder = 0
        output = []

        while (index < len(valL1)):
            val = valL1[index] + valL2[index] + remainder
            ones = val % 10
            output.append(ones)

            remainder = val // 10

            index += 1

        if remainder > 0:
            output.append(remainder)

        output.reverse()

        curr, prev = None, None

        for val in output:
            curr = ListNode(val, prev)
            prev = curr

        return curr


        

